certifikit/src/main/kotlin/app/cash/certifikit/text/pem.kt

```diff
@@ -0,0 +1,234 @@
+ * [rfc_7468]: https://tools.ietf.org/html/rfc7468
+ */
+fun PrivateKey.privateKeyPkcs1Pem(): String {
+  check(this is RSAPrivateKey) { "PKCS1 only supports RSA keys" }
+  return buildString {
+    append("-----BEGIN RSA PRIVATE KEY-----\n")
+    encodeBase64Lines(pkcs1Bytes())
+    append("-----END RSA PRIVATE KEY-----\n")
+  }
+}
+
+private fun PrivateKey.pkcs1Bytes(): ByteString {
+  val decoded = CertificateAdapters.privateKeyInfo.fromDer(this.encoded.toByteString())
+  return decoded.privateKey
+}
+
+/**
+ * Decodes a multiline string that contains both a [certificate][certificatePem] and a
+ * [private key][privateKeyPkcs8Pem], both [PEM-encoded][rfc_7468]. A typical input string looks
+ * like this:
+ *
+ * ```
+ * -----BEGIN CERTIFICATE-----
+ * MIIBYTCCAQegAwIBAgIBKjAKBggqhkjOPQQDAjApMRQwEgYDVQQLEwtlbmdpbmVl
+ * cmluZzERMA8GA1UEAxMIY2FzaC5hcHAwHhcNNzAwMTAxMDAwMDA1WhcNNzAwMTAx
+ * MDAwMDEwWjApMRQwEgYDVQQLEwtlbmdpbmVlcmluZzERMA8GA1UEAxMIY2FzaC5h
+ * cHAwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAASda8ChkQXxGELnrV/oBnIAx3dD
+ * ocUOJfdz4pOJTP6dVQB9U3UBiW5uSX/MoOD0LL5zG3bVyL3Y6pDwKuYvfLNhoyAw
+ * HjAcBgNVHREBAf8EEjAQhwQBAQEBgghjYXNoLmFwcDAKBggqhkjOPQQDAgNIADBF
+ * AiAyHHg1N6YDDQiY920+cnI5XSZwEGhAtb9PYWO8bLmkcQIhAI2CfEZf3V/obmdT
+ * yyaoEufLKVXhrTQhRfodTeigi4RX
+ * -----END CERTIFICATE-----
+ * -----BEGIN PRIVATE KEY-----
+ * MEECAQAwEwYHKoZIzj0CAQYIKoZIzj0DAQcEJzAlAgEBBCA7ODT0xhGSNn4ESj6J
+ * lu/GJQZoU9lDrCPeUcQ28tzOWw==
+ * -----END PRIVATE KEY-----
+ * ```
+ *
+ * The string should contain exactly one certificate and one private key in [PKCS #8][rfc_5208]
+ * format. It should not contain any other PEM-encoded blocks, but it may contain other text
+ * which will be ignored.
+ *
+ * Encode a held certificate into this format by concatenating the results of
+ * [certificatePem()][certificatePem] and [privateKeyPkcs8Pem()][privateKeyPkcs8Pem].
+ *
+ * [rfc_7468]: https://tools.ietf.org/html/rfc7468
+ * [rfc_5208]: https://tools.ietf.org/html/rfc5208
+ */
+fun decode(certificateAndPrivateKeyPem: String): Pair<KeyPair, X509Certificate> {
+  var certificatePem: String? = null
+  var pkcs8Base64: String? = null
+  var pkcs1Base64: String? = null
+  for (match in PEM_REGEX.findAll(certificateAndPrivateKeyPem)) {
+    when (val label = match.groups[1]!!.value) {
+      "CERTIFICATE" -> {
+        require(certificatePem == null) { "string includes multiple certificates" }
+        certificatePem = match.groups[0]!!.value // Keep --BEGIN-- and --END-- for certificates.
+      }
+      "PRIVATE KEY" -> {
+        require(pkcs8Base64 == null && pkcs1Base64 == null) { "string includes multiple private keys" }
+        pkcs8Base64 = match.groups[2]!!.value // Include the contents only for PKCS8.
+      }
+      "RSA PRIVATE KEY" -> {
+        require(pkcs8Base64 == null && pkcs1Base64 == null) { "string includes multiple private keys" }
+        pkcs1Base64 = match.groups[2]!!.value // Include the contents only for PKCS8.
```

Comment:
Is this a copy paste error?
```suggestion
        pkcs1Base64 = match.groups[2]!!.value // Include the contents only for PKCS1.
```